French Accent

brainfood1183 opened the pull request:
Added - French accent added.
Added - French Accent trait.
Added - Mime has a french accent by default (if mime breaks vow you will hear it).
Added - french beret (its black) and is in the theatrevending machine
Added - Wearing the french beret will give you a french accent.

French accent swaps specific words for their french equivalent. th is replaced with 'z and h at the beginning of a word is missing. It also changes the spacing between certain punctuation so this? this! this: and this; would be 'zis ? 'zis ! 'zis : and 'zis ;


https://github.com/space-wizards/space-station-14/assets/113240905/7b1583c2-beaa-4e5a-9f96-b8360952522c




- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase


**Changelog**
<!--
:cl:
- add: Some of the crew seem to be speaking with french accents (french accent added).
-->

Tynobeard commented:
Does is change bread to baguette for no reason? It should

brainfood1183 commented:
> Does is change bread to baguette for no reason? It should

it doesnt but baguette isnt french for bread :)

deltanedas commented:
tf2 spy

lapatison commented:
BTW, ClothingHeadHatBeret is also a [senior officer](https://github.com/space-wizards/space-station-14/blob/40afc2719b99b4af7a8071c7e6367827f72d9ded/Resources/Prototypes/Roles/Jobs/Security/senior_officer.yml#L40) default hat. Probably, something should be changed

CatMagic1 commented:
> > Does is change bread to baguette for no reason? It should
> 
> it doesnt but baguette isnt french for bread :)

and prosciutto isn't italian for meat

make bread baguette!!!

brainfood1183 commented:
> BTW, ClothingHeadHatBeret is also a [senior officer](https://github.com/space-wizards/space-station-14/blob/40afc2719b99b4af7a8071c7e6367827f72d9ded/Resources/Prototypes/Roles/Jobs/Security/senior_officer.yml#L40) default hat. Probably, something should be changed

yea i will sort that thanks for letting me know

TsjipTsjip commented:
Related PR: https://github.com/space-wizards/space-station-14/pull/21433

arimah commented on Content.Server/Speech/EntitySystems/FrenchAccentSystem.cs:
```diff
@@ -0,0 +1,38 @@
+        msg = _replacement.ApplyReplacements(msg, "french");
+
+        // replaces th with dz 
+        msg = Regex.Replace(msg, @"th", "dz", RegexOptions.IgnoreCase);
```
In my experience, native French speakers do not tend to turn all "th" into "dz"; it tends to become "z" or "s" depending on whether it was voiced or not.

Suggestion: don't mess with "th" unless the logic is made much smarter.

brainfood1183 replied:
i changed it to 'z i think this is a better overall solution. I dont think accent needs to be exactly perfect with all logic cases since its typically aimed towards english speaking servers.

arimah commented on Content.Server/Speech/EntitySystems/FrenchAccentSystem.cs:
```diff
@@ -0,0 +1,38 @@
+        msg = Regex.Replace(msg, @"th", "dz", RegexOptions.IgnoreCase);
+
+        // removes the letter h from the start of words.
+        msg = Regex.Replace(msg, @"(?<!\w)[h]", "", RegexOptions.IgnoreCase);
```
Suggestion: insert an apostrophe instead of deleting the letter, so one can tell a letter was omitted.

English as a large number of h-∅ minimal pairs (words that differ only in the presence of 'h'). This might make it ard to guess the original intent.

brainfood1183 replied:
this is now an apostrophe instead of deleting letter.

arimah reviewed (commented):
Salut! I'm not a maintainer, just a language nerd passing by, and I thought I'd take a look and leave some comments. :) In general, a decent effort!

A suggestion for a feature one could add, to make it properly French: insert a single space before every "?", "!", ":" and ";" preceded by a word character. While subtle, it plays into French orthographic tradition, where those punctuation marks (and more) have a space before _and_ after. The word character check would ensure that punctuation clusters like "!!" and "?!" are not broken apart - and of course actual French people can continue to insert their spaces manually to the confusion of everyone else without interference.

brainfood1183 commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,149 @@
+accent-french-replacement-44 = morte
+
+accent-french-replaced-45 = i
+accent-french-replacement-45 = je
```
yea will do

brainfood1183 commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,149 @@
+accent-french-replacement-10 = bien
+
+accent-nadsat-replaced-11 = great
+accent-nadsat-replacement-11 = super
```
fixed this issue.

brainfood1183 commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,149 @@
+accent-french-replacement-8 = amour
+
+accent-french-replaced-9 = your
+accent-french-replacement-9 = ton
```
removed these 2 replacements.

brainfood1183 commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,149 @@
+accent-french-replacement-18 = baguette
+
+accent-french-replaced-19 = magic
+accent-french-replacement-19 = la magie
```
omited the la from this replacement.

brainfood1183 commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,149 @@
+accent-french-replacement-14 = il
+
+accent-french-replaced-15 = her
+accent-french-replacement-15 = son
```
removed these replacements.

brainfood1183 commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,149 @@
+accent-french-replacement-7 = est
+
+accent-french-replaced-8 = love
+accent-french-replacement-8 = amour
```
removed this replacement.

brainfood1183 commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,149 @@
+accent-french-replacement-42 = bibliothécaire
+
+accent-french-replaced-43 = by
+accent-french-replacement-43 = par
```
removed this replacement

brainfood1183 commented:
> Salut! I'm not a maintainer, just a language nerd passing by, and I thought I'd take a look and leave some comments. :) In general, a decent effort!
> 
> A suggestion for a feature one could add, to make it properly French: insert a single space before every "?", "!", ":" and ";" preceded by a word character. While subtle, it plays into French orthographic tradition, where those punctuation marks (and more) have a space before _and_ after. The word character check would ensure that punctuation clusters like "!!" and "?!" are not broken apart - and of course actual French people can continue to insert their spaces manually to the confusion of everyone else without interference.

now if any of these are at the end of a word ? , ! such as this? this, this! they are now written this , this ? this !

Aisu9 commented:
Th- would be closer to z OR s than dz, i spent time in classes with a good amount of folks not talking english correctly.
so it would be zere and not dzere (why do you think we add a d? i know we are not the most proficient in english but god damn it there is a thing between talking a language incorrectly and adding letters were there should not be)

Aisu9 commented:
"r" are also much more articulated
such a gentleman would pronounce "where are you" - "were ar yu"

brainfood1183 commented:
> Th- would be closer to z OR s than dz, i spent time in classes with a good amount of folks not talking english correctly. so it would be zere and not dzere (why do you think we add a d? i know we are not the most proficient in english but god damn it there is a thing between talking a language incorrectly and adding letters were there should not be)

it is now 'z

potato1234x commented:
Oui oui hon hon baguette

lzk228 commented:
should it be a trait? italian accent isn't a trait

brainfood1183 commented:
> should it be a trait? italian accent isn't a trait

should italian accent be a trait?

UbaserB commented:
> > should it be a trait? italian accent isn't a trait
> 
> should italian accent be a trait?

i feel like there are already too many accent traits. Unless we get a specific “accent selector”, i feel like there’d be too much bloat.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-5 = goodbye
+accent-french-replacement-5 = au revoir
+
+accent-french-replaced-6 = the
```
This seems excessive. A French speaker would probably use the correct English article, especially since the French article is gendered and the 'le' or 'la' depends on the noun.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-6 = the
+accent-french-replacement-6 = la
+
+accent-french-replaced-7 = is
```
Also excessive and better handled by your regex rules.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-7 = is
+accent-french-replacement-7 = est
+
+accent-french-replaced-8 = bread
```
This seems too specific.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-10 = good
+accent-french-replacement-10 = bien
+
+accent-french-replaced-11 = great
```
I get what you're going for, but 'super' is going to sound like the English 'super' and its meaning is going to be lost.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-11 = great
+accent-french-replacement-11 = super
+
+accent-french-replaced-12 = beautiful
```
This seems too specific.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-12 = beautiful
+accent-french-replacement-12 = beau
+
+accent-french-replaced-13 = music
```
This seems too specific.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-14 = greetings
+accent-french-replacement-14 = bonjour
+
+accent-french-replaced-15 = france
```
This is overdoing it. This would change "I'm from France" to "I'm from la france" which a speaker would never do since it's "de France"

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-15 = france
+accent-french-replacement-15 = la france
+
+accent-french-replaced-16 = captain
```
This is overdoing it.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-16 = captain
+accent-french-replacement-16 = capitaine
+
+accent-french-replaced-17 = fries
```
This is overdoing it.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-17 = fries
+accent-french-replacement-17 = frites
+
+accent-french-replaced-18 = wand
+accent-french-replacement-18 = baguette
+
+accent-french-replaced-19 = magic
+accent-french-replacement-19 = magie
+
+accent-french-replaced-20 = magic wand
+accent-french-replacement-20 = baguette magique
+
+accent-french-replaced-21 = we
+accent-french-replacement-21 = nous
```
This is overdoing it.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-23 = thanks
+accent-french-replacement-23 = merci
+
+accent-french-replaced-24 = thankyou
+accent-french-replacement-24 = merci
```
Why is this missing a space? Are you trying to catch typos too? This is overdoing it.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replacement-27 = sécurité
+
+accent-french-replaced-28 = kitchen
+accent-french-replacement-28 = cuisine
```
This would be misread by English speakers.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-28 = kitchen
+accent-french-replacement-28 = cuisine
+
+accent-french-replaced-29 = quick
+accent-french-replacement-29 = rapide
+
+accent-french-replaced-30 = library
+accent-french-replacement-30 = bibliothèque
```
This is overdoing it.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-31 = shit
+accent-french-replacement-31 = merde
+
+accent-french-replaced-32 = if
```
This would make it too hard to understand.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-32 = if
+accent-french-replacement-32 = si
+
+accent-french-replaced-33 = traitor
+accent-french-replacement-33 = traitre
+
+accent-french-replaced-34 = murdered
+accent-french-replacement-34 = assassiné
+
+accent-french-replaced-35 = botany
+accent-french-replacement-35 = botanique
+
+accent-french-replaced-36 = murder
+accent-french-replacement-36 = meurtre
+
+accent-french-replaced-37 = banana
+accent-french-replacement-37 = banane
```
This is overdoing it.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-39 = please
+accent-french-replacement-39 = s'il vous plaît
+
+accent-french-replaced-40 = will you
+accent-french-replacement-40 = veux-tu
```
This is overdoing it.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-42 = librarian
+accent-french-replacement-42 = bibliothécaire
+
+accent-french-replaced-43 = yo
```
What?

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-41 = french
+accent-french-replacement-41 = français
+
+accent-french-replaced-42 = librarian
+accent-french-replacement-42 = bibliothécaire
```
Too specific.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
```diff
@@ -0,0 +1,131 @@
+accent-french-replaced-30 = library
+accent-french-replacement-30 = bibliothèque
+
+accent-french-replaced-31 = shit
```
French people know how to say the English word "shit".

Partmedia commented on Resources/Prototypes/Entities/Clothing/Head/hats.yml:
```diff
@@ -25,6 +25,24 @@
+  parent: ClothingHeadBase
+  id: ClothingHeadHatBeretFrench
+  name: french beret
+  description: A french beret, viva la france.
```
English capitalization

Partmedia reviewed (changes requested):
Thank you for your contribution. Overall, I think you've overdone the word replacements, fixating too much on specific words. I would rather focus on more systematic regex replacements for phonemes that French speakers are most likely to mispronounce. This is an accent system, not a translation system.

Partmedia commented:
Hi, thanks for updating this branch. My comments about a lot of these replacement words still stands. I would prefer to keep them simple and do most of the work in [phoneme](https://en.wikipedia.org/wiki/Phoneme) replacements.

brainfood1183 commented:
> Hi, thanks for updating this branch. My comments about a lot of these replacement words still stands. I would prefer to keep them simple and do most of the work in [phoneme](https://en.wikipedia.org/wiki/Phoneme) replacements.

Yea np, i think i removed them all now and i also removed french accent from the traits list.

Partmedia commented on Resources/Locale/en-US/accent/french.ftl:
I still think there are too many hard-coded words. There are some basic sentences that break:

- "Please don't do this" -> "s'il vous plaît don't do this" but it should be "don't do this s'il vous plaît"

These sort of errors make this feel hackish and low-effort. Real speakers also don't mix up words from different languages.

brainfood1183 replied:
I removed the word replacements.

tutoumi commented:
Finally I'll be able to write proper english without caring about Gordon's IC french accent that i can't do because i am too concentrated writing english without errors (i am french). also, there is a problem with the le and la, don't always put la cause its for women or feminin objects, la porte (the door), la baguette (the baguette) and le for the men and the masculin objects, le pain (the bread), le clown (the clown). I don't know how you can fix this so good luck.

brainfood1183 commented:
https://github.com/space-wizards/space-station-14/assets/113240905/88f41e2d-1787-4d08-87b4-9075e8c12bf3

tutoumi commented:
I find this better personaly 👍

Partmedia commented on Content.Server/Speech/EntitySystems/FrenchAccentSystem.cs:
```diff
@@ -0,0 +1,43 @@
+        SubscribeLocalEvent<FrenchAccentComponent, AccentGetEvent>(OnAccentGet);
+    }
+
+    // converts left word when typed into the right word. For example typing you becomes ye.
```
This comment looks copy/pasted and isn't accurate.

Partmedia reviewed (commented):
This mostly looks good to me now. Would you mind fixing the last few minor nits?

tutoumi commented on Resources/Prototypes/Entities/Clothing/Head/hats.yml:
```diff
@@ -25,6 +25,24 @@
+- type: entity
+  parent: ClothingHeadBase
+  id: ClothingHeadHatBeretFrench
+  name: french beret
+  description: A french beret, viva la France.
```
''viva'' la France ?

brainfood1183 replied:
i dont think we ever capitalize names of things. but i can modify the description.

Partmedia replied:
All item names are in lower case *unless* there is a proper noun, in which case the proper noun should be capitalized.

brainfood1183 replied:
made these changes.

brainfood1183 commented on Resources/Locale/en-US/traits/traits.ftl:
```diff
@@ -24,6 +24,9 @@ trait-unrevivable-desc = You are unable to be revived by defibrillators.
 trait-pirate-accent-desc = You can't stop speaking like a pirate!
 
+trait-french-accent-name = French Accent
+trait-french-accent-desc = You like to throw random french words into your english out of protest.
```
This was removed as french accent is no longer a trait (advised not add more accent traits).

Partmedia commented:
Sorry to bug you once more. There is a current freeze on pet-wearable sprites: https://github.com/space-wizards/space-station-14/issues/8524

Please remove the hamlet sprite.

brainfood1183 commented:
> Sorry to bug you once more. There is a current freeze on pet-wearable sprites: #8524
> 
> Please remove the hamlet sprite.

sure, its done.

Partmedia commented:
Thank you for your contribution and for your effort.
